dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -414,4 +424,50 @@ public Config getConfig() {
         }
     }
 
+    @GET
+    @Timed
+    @Path("cli-info")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Get Dockstore CLI information.", description = "Get Dockstore CLI information. NO authentication")
+    @ApiOperation(value = "Get Dockstore CLI information.", notes = "Get Dockstore CLI information. NO authentication")
+    public CLIInfo getCliVersion() {
+        GHRelease ghRelease;
+        PagedIterable<GHRelease> allReleases;
+        CLIInfo cliInfo = new CLIInfo();
+        try {
+            GitHub github = new GitHubBuilder().build();
+            GHRepository repository = github.getRepository("dockstore/dockstore-cli");
+            ghRelease = repository.getLatestRelease();
+            allReleases = repository.listReleases();
+        } catch (IOException e) {
+            LOG.info("Could not get CLI releases information from GitHub. ", e);
+            throw new CustomWebApplicationException("Could not get CLI  releases information info from GitHub",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        }
+        String cliLatestVersion = ghRelease.getName();
+        cliInfo.setCliLatestDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestVersion + "/dockstore");
+        cliInfo.setCliLatestVersion(cliLatestVersion);
+        // Get the latest unstable version
+        // If there is no unstable version return an empty string as the latest unstable version
+        String cliLatestUnstableVersion = getLatestUnstableVersion(allReleases).orElse("");
+        cliInfo.setCliLatestUnstableDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestUnstableVersion + "/dockstore");
+        cliInfo.setCliLatestUnstableVersion(cliLatestUnstableVersion);
+        return cliInfo;
+    }
+
+    /**
+     * This method will return the latest unstable version
+     * Assumes the first unstable version in the array is the latest unstable version
+     *
+     * @param allReleases a PagedIterable of GHRelease objects for releases
+     * @return An Optional String of the latest unstable version
+     */
+    private static Optional<String> getLatestUnstableVersion(PagedIterable<GHRelease> allReleases) {
+        try {
+            return Optional.of(Arrays.stream(allReleases.toArray()).filter(GHRelease::isPrerelease).findFirst().orElseThrow().getName());
```

Comment:
I don't think you want to do `toArray` --> JavaDoc says it will`Eagerly walk {@link PagedIterator} and return the result in an array.`, i.e., it will force all releases to fetched before starting to look for the latest, when the latest is going to be close to the top.

Reply:
Now iterates over pages and checks each page group